pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -172,15 +172,18 @@ func NewFlowExporter(podStore podstore.Interface, proxier proxy.Proxier, k8sClie
 		return nil, err
 	}
 	expInput := prepareExporterInputArgs(o.FlowCollectorProto, nodeName)
-	var l7Listener *connections.L7Listener
-	if l7FlowExporterEnabled {
-		l7Listener = connections.NewL7Listener(podL7FlowExporterAttrGetter, podStore)
-	}
 
 	connTrackDumper := connections.InitializeConnTrackDumper(nodeConfig, serviceCIDRNet, serviceCIDRNetv6, ovsDatapathType, proxyEnabled)
 	denyConnStore := connections.NewDenyConnectionStore(podStore, proxier, o)
-	conntrackConnStore := connections.NewConntrackConnectionStore(connTrackDumper, v4Enabled, v6Enabled, npQuerier, podStore, proxier, l7Listener, o)
 
+	var l7Listener *connections.L7Listener
+	var conntrackConnStore *connections.ConntrackConnectionStore
+	if l7FlowExporterEnabled {
+		l7Listener = connections.NewL7Listener(podL7FlowExporterAttrGetter, podStore)
+		conntrackConnStore = connections.NewConntrackConnectionStore(connTrackDumper, v4Enabled, v6Enabled, npQuerier, podStore, proxier, l7Listener, o)
+	} else {
+		conntrackConnStore = connections.NewConntrackConnectionStore(connTrackDumper, v4Enabled, v6Enabled, npQuerier, podStore, proxier, nil, o)
+	}
```

Comment:
you can also avoid duplicating the call to `conntrackConnStore` if you want:

```go
var l7Listener *connections.L7Listener
var eventMapGetter l7EventMapGetter
if l7FlowExporterEnabled {
    l7Listener = connections.NewL7Listener(podL7FlowExporterAttrGetter, podStore)
    eventMapGetter = l7Listener
}
conntrackConnStore := connections.NewConntrackConnectionStore(connTrackDumper, v4Enabled, v6Enabled, npQuerier, podStore, proxier, eventMapGetter, o)
```

Reply:
Updated. Thanks!